src/ui/pages.rs

```diff
@@ -1,12 +1,21 @@
 use gtk4::{prelude::*, Button, Stack};
+use gtk4::glib::{self, clone};
 use gtk4::{Entry, Orientation};
 use std::rc::Rc;
-use std::{fs::File, io::Write};
+use std::{sync::OnceLock, fs::File, io::Write};
+use std::sync::{Arc, mpsc};
+use reqwest::header::{HeaderMap, HeaderValue};
+use tokio::runtime::Runtime;
+use crate::{get_tokens, runtime};
 
 use crate::LoginInfo;
 
 pub fn login_page(parent_stack: Rc<Stack>) {
+
+
+
     let login = gtk4::Box::new(Orientation::Vertical, 5);
+    let url = "https://discord.com/api/v9/users/@me";
```

Comment:
Feels like this should be broken down to into a var and a const.
`https://discord.com/api/v9` and the specific end point.